index.bs

```diff
@@ -337,59 +296,136 @@ named |fragment| from a Sanitizer |input|, run these steps:
+  3. if m[node] is `keep`, do nothing.
 
-1. run [=create a document fragment=] algorithm on the |input|.
-2. run the [=sanitize document fragment=] algorithm on the resulting fragment,
-3. and return its result.
+To <dfn>sanitize a node</dfn> named |node| run these steps:
 
-To <dfn>sanitizeToString</dfn> a given |input|, run these steps:
+1. let |config| be the Sanitizer's [=effective configuration=].
+2. if |node| is an element node:
+  1. let |element| be |node|'s element.
+  2. [=list/iterate|for each=] |attr| in |element|'s [=Element/attribute list=]:
+    1. determine the [=sanitize action=] that |config| assigns to the |element| and |attr| pair.
+    2. if the result is different from `keep`, remove |attr| from |element|.
+   3. return the [=sanitize action=] that |config| assigns to |element|.
+3. otherwise, return 'keep'
+
+Issue: What about comment nodes, CDATA, etc. ?
+
+### The Effective Configuration {#configuration}
+
+A Sanitizer is potentially complex, so we will define a helper
+construct, the *effective configuration*. This is mostly a specification
+convenience and allows us to explain a Sanitizer's operation in two steps:
+One, how to derive the effective configuration, and two, define the
+Sanitzer's operation based on it.
+
+An <dfn>effective configuration</dfn> maps a given |element| or a given pair of
+|element| and |attribute| to a [=sanitize action=].
+A <dfn>sanitize action</dfn> can have the values `keep`, `drop`, or `block`.
+
+A Sanitizer's [=effective configuration=] is merged from the
+[=baseline effective configuration=] and the effective configuration derived
+from the Sanitizer's [=configuration object=]. If no configuration object has
+been provided, the built-in [=default configuration=] is used instead.
+To merge two (or more)
+[=effective configurations=], map any given |element| or a pair of |element|
+and |attribute| to the [=stricter action=] of its constituent configurations.
+To determine the <dfn>stricter action</dfn> of two [=sanitize actions=], pick
+the 'larger' of the two actions assuming a transitively defined order with
+`drop` &gt; `block`, and `block` &gt; `keep`.
+
+Note: This definition of stricter actions ensures that the built-in baseline
+      configuration cannot be overriden, and therefor forms a hard guarantee
+      for all Sanitizer instances.
+
+Before describing how an effective configuration is derived, we need a
+helper definition:  The <dfn>element kind</dfn> of an |element| is one of
+`regular`, `unknown`, or `custom`. Let |kind| be:
+- `custom`, if |element|'s tag name is a [=valid custom element name=],
+- `unknown`, if |element| is not in the [[HTML]] namespace or if |element|'s
+   tag name denotes an unknown element &mdash; that is, if the
+   [=element interface=] the [[HTML]] specification assigns to it would
+   be {{HTMLUnknownElement}},
+- `regular`, otherwise.
+
+Similarly, the <dfn>attribute kind</dfn> of an |attribute| is one of `regular`
```

Comment:
As I read it, `<data>` is `regular` (because of the HTMLDataElement interface), `<data-foo>` would be `custom`.

I think HTML spec doesn't have `<data-*>` elements , but it does have `data-*=` attributes. But that means we just have the same problem just a few lines down. :)  I'd think that `data-foo=` would also be `regular`, although that isn't as nicely defined.

Note that in this PR I've dropped the idea of an `allow-unknown:` option, so there's a blanket allow-all option only for custom elements, not for unknowns.

Reply:
Sorry, this was specifically meant for.